### PR TITLE
Indexer Request torrentz2.eu

### PR DESCRIPTION
**Pull request by scambra**
I added this definition for torrentz2.eu, I don't know if I did categories correctly or some important fields are missing, but it's working for me

For a new indexer, please follow this checklist:

- [ ] Run `cardigann test definitions/trackername.yml` and include the output here:

```
→ Testing 1 definition(s) (1.9.9/linux/amd64)
→ Testing indexer torrentz2 at https://torrentz2.eu/
  Testing search mode "search" SUCCESS ✓ in 662.565889ms
  Testing search mode "tv-search" SUCCESS ✓ in 154.924911ms
  Testing empty results are handled SUCCESS ✓ in 241.469645ms
  Testing ratio SUCCESS ✓ in 8.732µs
→ Indexer torrentz2 is OK
```

### CODE REVIEW

File: definitions/torrentz2.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  caps:
+    categories:
+      "Video Tv": TV
+      "Video": Movies
+      "Video Movie HD": Movies/HD
+
+    modes:
+      search: [f]
+      tv-search: [f, season, ep]
```
Comment (kaso17) [inline definitions/torrentz2.yml]: these parameters refer to the newznab API, f needs to be replaced with q

File: definitions/torrentz2.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        filters:
+          - name: regexp
+            args: "Hash: (\\w+)"
+      category:
+        selector: category
+      date:
+        selector: pubDate
+        filters:
+          - name: dateparse
+            args: "Mon, 02 Jan 2006 15:04:05 -0700"
```
Comment (kaso17) [inline definitions/torrentz2.yml]: this doesn't cover all cases.
day and hours seem to be used without leading zeros.
The problem is that go doesn't support the 24 hour format without leading space.
A workaround would be to use "Mon, 2 Jan 2006 15:04:05 -0700" but use 9 replace filters like [" 1:", " 01:"] before that to convert the 12H format to 24H.

Comment (scambra) [inline definitions/torrentz2.yml]: I fixed all these issues

**Comment by scambra** [comment]
I don't see how to get latest torrents with feed, using normal search. Why does it matter? For testing?

I didn't know what field were seeders and peers, was clearer in feed, but comparing search and feed results I get them:

```html
<dl><dt><a href=/7e1faf6ee57b3f2961dc02c5ef01ba809e464fcb>The.Expanse.S02E02.1080p.WEB-DL.DD5.1.H264-RARBG</a> &#187; video tv</dt><dd><span>✓</span><span title=1486037095>6 days</span><span>1665 MB</span><span>12</span><span>3</span></dd></dl>
```

from feed, it has 12 seeds and 15 peers, so needs to sum lasts 2 span for peers, is it possible? category may be harder

For date, I don't see how to get from unix timestamp using title attribute, I tried to use timeago with span text but it's not working, I get:
DEBU[0000] Applying filter timeago                       args=<nil> before=1 day site=torrentz2-html

cardigann: error: One or more tests failed, try --help

**Comment by scambra** [comment]
https://gist.github.com/scambra/287321c56c7c8fe57704dead4b12268c

File: definitions/torrentz2.yml (hunk 1 above)
Comment (scambra) [inline definitions/torrentz2.yml]: Ok, I wasn't sure about it but I didn't found doc about creating definitions, I have changed to q but I don't see any difference

**Comment by kaso17** [comment]
Not sure what's your final use case but e.g. sonarr and radarr normally only grab the latest releases from the indexers and scan them for matching shows/movies (q/f would be empty). Only if you do a manual search it will search for a specific torrent. That's why the the latest release should be returned in case no search string is provided.

Regarding your date issue:
timestamp parsing is not supported by cardigann (see #208)
To make the span text parsing work you've to append " ago" via a filter.

The bigger problem with torrentz2 seems to be the download link. See https://github.com/Jackett/Jackett/pull/1042

**Comment by scambra** [comment]
I think I had downloaded link from torrentz and it worked, at least using feed, I will double check
My use case is sonarr, but I have been testing only with manual search. I think html definition is mostly working, only issue is I'm getting leetchs as peers instead of sum of seeds+leetchs

Is it possible to sum seeds and leetchs to get peers? feed has seeds and peers, but search returns seeds and leetchs

**Comment by kaso17** [comment]
The "peers" value from torrentz seems to be just the leechers, just assign it to the "leechers" field. Cardigann will automatically sum seeders+leechers.

With the feed implementation you're extracting the infohash. Cardigann adds the base url, resulting in http://torrentz2.eu/$INFO_HASH as download link. But that's just the details page, download link should be a link to a torrent file or a magnet link.

**Comment by scambra** [comment]
download is not working, I thought I saw working, but it isn't
I was using leechers, so definition for feed was wrong. Anyway, I will replace with search url, and fix download building magnet link from hash.

**Comment by kaso17** [comment]
merged, thank you.
Would you consider adding a few common trackers to the magnet url a good idea so we don't have to solely rely on DTH?

**Comment by scambra** [comment]
I have no problem downloading these torrents without tracker info, maybe because I have some other torrents and getting info from DHT quickly, I don't know, but it could be good idea

**Comment by ninovanhooff** [comment]
I'd like to use this indexer, but I don't see it in the available options. Even after updating to edge 1.9.9-11-ga1ab309

I guess edge channel is not up to date with master yet?

**Pull request merged**